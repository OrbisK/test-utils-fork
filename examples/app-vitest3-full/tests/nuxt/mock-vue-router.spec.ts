import { describe, expect, it, vi } from 'vitest'
import { mockNuxtImport, mountSuspended } from '@nuxt/test-utils/runtime'
import Index from '~/pages/router/route.vue'

vi.mock('vue-router', () => ({
  useRoute: vi.fn(() => ({
    meta: {},
    path: '/123',
    query: {},
  })),
}))

mockNuxtImport('useRoute', () => () => ({
  meta: {},
  path: '/bob',
  query: {},
}))

describe('Index', async () => {
  const wrapper = await mountSuspended(Index)

  it('should render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
