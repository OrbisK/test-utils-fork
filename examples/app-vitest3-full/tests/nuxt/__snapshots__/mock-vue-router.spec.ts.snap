// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Index > should render correctly 1`] = `"<div>Index page, path: /123, /bob</div>"`;
